www/install/steps/process/createDbUser.php

```
@@ -82,48 +82,132 @@
+    $findUserStatement->bindValue(':dbUser', $queryValues[':dbUser'], \PDO::PARAM_STR);
+    $findUserStatement->bindValue(':host', $queryValues[':host'], \PDO::PARAM_STR);
+    $findUserStatement->execute();
+    // If user doesn't exist, create it
+    if ($result = $findUserStatement->fetch(\PDO::FETCH_ASSOC) === false) {
+        $prepareCreate = $link->prepare($createUser);
+        $prepareAlter = $link->prepare($alterQuery);
+        foreach ($queryValues as $key => $value) {
+            $prepareCreate->bindValue($key, $value, \PDO::PARAM_STR);
+            $prepareAlter->bindValue($key, $value, \PDO::PARAM_STR);
         }
-    }
-    if ((strpos($versionName, "MariaDB") !== false && version_compare($versionNumber, '10.2.0') >= 0)
-        || (strpos($versionName, "MySQL") !== false && version_compare($versionNumber, '8.0.0') >= 0)
-    ) {
-        // altering the mysql's password plugin using the ALTER USER request
-        $prepareAlter->execute();
-    }
+        // creating the user
+        $prepareCreate->execute();
 
-    // granting privileges
-    $link->exec(sprintf($query, $parameters['db_configuration']));
-    $link->exec(sprintf($query, $parameters['db_storage']));
+        // checking mysql version before trying to alter the password plugin
+        $prepareCheckVersion = $link->query($checkMysqlVersion);
+        $versionName = $versionNumber = "";
+        while ($row = $prepareCheckVersion->fetch()) {
+            if ($row['Variable_name'] === "version") {
+                $versionNumber = $row['Variable_name'];
+            } elseif ($row['Variable_name'] === "version_comment") {
+                $versionName = $row['Variable_name'];
+            }
+        }
+        if (
+            (strpos($versionName, "MariaDB") !== false && version_compare($versionNumber, '10.2.0') >= 0)
+            || (strpos($versionName, "MySQL") !== false && version_compare($versionNumber, '8.0.0') >= 0)
```

Should we throw an exception in case database version does not match those conditions ?

---

we let this to avoid side effects